pkg/utils/peers_test.go

```diff
@@ -0,0 +1,62 @@
+package utils
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Utils/Peers tests", func() {
+
+	var nrOfNodes int
+
+	Context("small cluster", func() {
+		BeforeEach(func() {
+			nrOfNodes = 2
+		})
+		It("GetNextBatchSize should return correct first batch size", func() {
+			Expect(GetNextBatchSize(nrOfNodes, nrOfNodes)).To(Equal(2))
+		})
+		It("GetNrOfBatches should return correct nr of batches", func() {
+			Expect(GetNrOfBatches(nrOfNodes)).To(Equal(1))
+		})
+	})
+
+	Context("medium cluster", func() {
+		BeforeEach(func() {
+			nrOfNodes = 20
+			// expected batches: 3 + 5*3 + 2 = 7 batches
+		})
+		It("GetNextBatchSize should return correct first batch size", func() {
+			Expect(GetNextBatchSize(nrOfNodes, nrOfNodes)).To(Equal(3))
+		})
+		It("GetNextBatchSize should return correct next batch size", func() {
+			Expect(GetNextBatchSize(nrOfNodes, nrOfNodes-3)).To(Equal(3))
+		})
+		It("GetNextBatchSize should return correct last batch size", func() {
+			Expect(GetNextBatchSize(nrOfNodes, 2)).To(Equal(2))
+		})
```

Comment:
Nit: I think it'll be nicer to replace those tests with something like:
```
nodesLeft := nrOfNodes
expectedBatchesSize := []int{3, 3, 3, 3, 3, 3, 2, 0}
for _, expectedBatchSize := range expectedBatchesSize {
	actualBatchSize := GetNextBatchSize(nrOfNodes, nodesLeft)
	Expect(actualBatchSize).To(Equal(expectedBatchSize))
	nodesLeft -= actualBatchSize
}
```
it can even be simplified farther by initializing nodesLeft in a JustBeforeEach block shared for all the tests in this file.

Reply:
good idea, done